src/MoBi.Presentation/Tasks/ModuleLoader.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+using System.Linq;
+using MoBi.Assets;
+using MoBi.Core.Commands;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Helper;
+using MoBi.Presentation.Tasks.Interaction;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Presentation.Tasks
+{
+   public interface IModuleLoader
+   {
+      IMoBiCommand LoadModule(MoBiProject parent);
+   }
+
+   public class ModuleLoader : IModuleLoader
+   {
+      private readonly IInteractionTasksForModule _moduleTask;
+      private readonly IInteractionTasksForExpressionProfileBuildingBlock _expressionTask;
+      private readonly IInteractionTasksForIndividualBuildingBlock _individualTask;
+      private readonly IInteractionTaskContext _interactionTaskContext;
+
+      public ModuleLoader(IInteractionTasksForModule moduleTask, IInteractionTasksForExpressionProfileBuildingBlock expressionTask, IInteractionTasksForIndividualBuildingBlock individualTask, IInteractionTaskContext interactionTaskContext)
+      {
+         _moduleTask = moduleTask;
+         _expressionTask = expressionTask;
+         _individualTask = individualTask;
+         _interactionTaskContext = interactionTaskContext;
+      }
+
+      public IMoBiCommand LoadModule(MoBiProject parent)
+      {
+         var filename = _moduleTask.AskForPKMLFileToOpen();
+
+         if (filename.IsNullOrEmpty())
+            return new MoBiEmptyCommand();
+
+         try
+         {
+            // If the user is adding modules from a simulation transfer, optionally add other building blocks
+            return addFromSimulationTransfer(parent, filename);
+         }
+         catch
+         {
+            // If the user is not loading from a simulation transfer, only add modules
+            return _moduleTask.AddItemsToProjectFromFile(filename, parent, null);
```

Comment:
what;s with the null here? explain

Reply:
That's part of the interaction tasks stuff. Formerly it was protected, but now that it's public, I'll make it optional as you suggest